--- cases/repo_41d48d9_5577479/merge_label.py	2026-03-03 00:15:28
+++ cases/repo_41d48d9_5577479/llm_label.py	2026-05-09 21:17:33
@@ -807,7 +807,6 @@
         parent directory where the conll_03 folder resides.
         If using entity linking, the conll03 dateset is reduced by about 20 Documents, which are not part of the yago dataset.
         :param base_path: Path to the CoNLL-03 corpus (i.e. 'conll_03' folder) on your machine
-        POS tags or chunks respectively
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
         """
@@ -852,7 +851,6 @@
         'conll_03_german'. Then set the base_path parameter in the constructor to the path to the parent directory where
         the conll_03_german folder resides.
         :param base_path: Path to the CoNLL-03 corpus (i.e. 'conll_03_german' folder) on your machine
-        word lemmas, POS tags or chunks respectively
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
         """
@@ -899,7 +897,6 @@
         download the dataset.
         :param base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
         to point to a different folder but typically this should not be necessary.
-        POS tags instead
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
         """
@@ -1295,7 +1292,6 @@
         download the dataset.
         :param base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
         to point to a different folder but typically this should not be necessary.
-        POS tags instead
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
         """
@@ -1407,7 +1403,6 @@
         in BIO format. The first time you call this constructor it will automatically download the dataset.
         :param base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
         to point to a different folder but typically this should not be necessary.
-        POS tags instead
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         """
         # column format
@@ -1452,7 +1447,6 @@
         in BIO format. The first time you call this constructor it will automatically download the dataset.
         :param base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
         to point to a different folder but typically this should not be necessary.
-        POS tags instead
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         """
         # column format
@@ -1490,7 +1484,6 @@
     Initialize corpus of SEC-fillings annotated with English NER tags. See paper "Domain Adaption of Named Entity
     Recognition to Support Credit Risk Assessment" by Alvarado et al, 2015: https://aclanthology.org/U15-1010/
     :param base_path: Path to the CoNLL-03 corpus (i.e. 'conll_03' folder) on your machine
-    POS tags or chunks respectively
     :param in_memory: If True, keeps dataset in memory giving speedups in training.
     :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
     """
@@ -1544,7 +1537,6 @@
         The first time you call this constructor it will automatically download the dataset.
         :param base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
         to point to a different folder but typically this should not be necessary.
-        POS tags instead
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
         """
@@ -1587,7 +1579,6 @@
         download the dataset.
         :param base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
         to point to a different folder but typically this should not be necessary.
-        POS tags instead
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
         """
@@ -1784,7 +1775,6 @@
         The first time you call this constructor it will automatically download the dataset.
         :param base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
         to point to a different folder but typically this should not be necessary.
-        POS tags instead
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
         """
@@ -2265,7 +2255,6 @@
         dataset.
         :param base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
         to point to a different folder but typically this should not be necessary.
-        POS tags instead
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
         """
@@ -2382,7 +2371,6 @@
         download the dataset.
         :param base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
         to point to a different folder but typically this should not be necessary.
-        POS tags instead
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
         """
@@ -2432,7 +2420,6 @@
         download the dataset.
         :param base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
         to point to a different folder but typically this should not be necessary.
-        POS tags instead
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
         """
@@ -2569,7 +2556,6 @@
         with the languages you require. If you pass "all", all languages will be initialized.
         :param base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
         to point to a different folder but typically this should not be necessary.
-        POS tags instead
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         """
         if not base_path:
@@ -2660,7 +2646,6 @@
         parent directory where the multiconer folder resides. You can also create the multiconer in
         the {FLAIR_CACHE_ROOT}/datasets folder to leave the path empty.
         :param base_path: Path to the CoNLL-03 corpus (i.e. 'conll_03' folder) on your machine
-        POS tags or chunks respectively
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         :param use_dev_as_test: If True, it uses the dev set as test set and samples random training data for a dev split.
         :param task: either 'multi', 'code-switch', or the language code for one of the mono tasks.
@@ -2753,8 +2738,6 @@
         base_path : Union[str, Path], optional
             Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
             to point to a different folder but typically this should not be necessary.
-            The data is in bio-format. It will by default (with the string "ner" as value) be transformed
-            into the bioes format. If you dont want that set it to None.
 
         """
         if type(languages) == str:
@@ -3176,8 +3159,6 @@
         base_path : Union[str, Path], optional
             Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
             to point to a different folder but typically this should not be necessary.
-            The data is in bio-format. It will by default (with the string "ner" as value) be transformed
-            into the bioes format. If you dont want that set it to None.
 
         """
         # if no languages are given as argument all languages used in XTREME will be loaded
@@ -3491,7 +3472,6 @@
         download the dataset.
         :param base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
         to point to a different folder but typically this should not be necessary.
-        POS tags instead
         :param in_memory: If True, keeps dataset in memory giving speedups in training.
         :param document_as_sequence: If True, all sentences of a document are read into a single Sentence object
         """
@@ -4085,15 +4065,6 @@
         :language: Language for a supported dataset.
         :base_path: Default is None, meaning that corpus gets auto-downloaded and loaded. You can override this
         to point to a different folder but typically this should not be necessary.
-        :in_memory: If True, keeps dataset in memory giving speedups in training.
-        :version: Version of CLEF-HIPE dataset. Currently only v1.0 is supported and available.
-        :branch_name: Defines git branch name of HIPE data repository (main by default).
-        :dev_split_name: Defines default name of development split (dev by default). Only the NewsEye dataset has
-        currently two development splits: dev and dev2.
-        :add_document_separator: If True, a special document seperator will be introduced. This is highly
-        recommended when using our FLERT approach.
-        :sample_missing_splits: If True, data is automatically sampled when certain data splits are None.
-        :preproc_fn: Function that is used for dataset preprocessing. If None, default preprocessing will be performed.
         """
         if not base_path:
             base_path = flair.cache_root / "datasets"
