diff --git a/salt/states/pkg.py b/salt/states/pkg.py
index 3ba9d3ff2c..3dc52901fb 100644
--- a/salt/states/pkg.py
+++ b/salt/states/pkg.py
@@ -465,7 +465,9 @@ def latest(
         desired_pkgs = [name]
 
     cur = __salt__['pkg.version'](*desired_pkgs)
-    avail = __salt__['pkg.available_version'](*desired_pkgs)
+    avail = __salt__['pkg.available_version'](*desired_pkgs,
+                                              fromrepo=fromrepo,
+                                              **kwargs)
 
     # Repack the cur/avail data if only a single package is being checked
     if isinstance(cur, basestring):
