--- cases/repo_5019c3b_3428d97/merge_pipeline.py	2026-03-28 21:38:48
+++ cases/repo_5019c3b_3428d97/llm_pipeline.py	2026-03-28 21:43:05
@@ -833,7 +833,7 @@
                                           center_traj)  # visualize
                 writer.write(im)
                 if self.file_name is None:  # use camera_id
-                    cv2.imshow('Paddle-Pipeline', im)
+                    cv2.imshow('PPHuman', im)
                     if cv2.waitKey(1) & 0xFF == ord('q'):
                         break
 
@@ -1003,4 +1003,4 @@
     assert FLAGS.device in ['CPU', 'GPU', 'XPU'
                             ], "device should be CPU, GPU or XPU"
 
-    main()
+    main()
\ No newline at end of file
