--- cases/repo_9ca574a_a96de0d/merge_keystone.py	2026-02-25 15:37:07
+++ cases/repo_9ca574a_a96de0d/llm_keystone.py	2026-02-25 16:41:56
@@ -45,35 +45,37 @@
 
     Only intended to be used within Keystone-enabled modules
     '''
-    user = __salt__['config.option']('keystone.user', 'admin')
-    password = __salt__['config.option']('keystone.password', 'ADMIN')
-    tenant = __salt__['config.option']('keystone.tenant', 'admin')
+    user = __salt__['config.option']('keystone.user')
+    password = __salt__['config.option']('keystone.password')
+    tenant = __salt__['config.option']('keystone.tenant')
     tenant_id = __salt__['config.option']('keystone.tenant_id')
-    auth_url = __salt__['config.option']('keystone.auth_url',
-                                         'http://127.0.0.1:35357/v2.0/')
-    insecure = __salt__['config.option']('keystone.insecure', False)
-    token = __salt__['config.option']('keystone.token', 'ADMIN')
-    endpoint = __salt__['config.option']('keystone.endpoint',
-                                         'http://127.0.0.1:35357/v2.0')
+    auth_url = __salt__['config.option']('keystone.auth_url')
+    insecure = __salt__['config.option']('keystone.insecure')
+    token = __salt__['config.option']('keystone.token')
+    endpoint = __salt__['config.option']('keystone.endpoint')
 
     kwargs = {}
     if token:
-        kwargs = {'token': token,
-                  'endpoint': endpoint}
+        kwargs = {
+                'token': token,
+                'endpoint': endpoint,
+                }
     else:
-        kwargs = {'username': user,
-                  'password': password,
-                  'tenant_name': tenant,
-                  'tenant_id': tenant_id,
-                  'auth_url': auth_url,
-                  'insecure': insecure}
+        kwargs = {
+                'username': user,
+                'password': password,
+                'tenant_name': tenant,
+                'tenant_id': tenant_id,
+                'auth_url': auth_url,
+                'insecure': insecure,
+                }
 
     return client.Client(**kwargs)
 
 
-def ec2_credentials_get(user_id=None,
+def ec2_credentials_get(user_id=None,       # pylint: disable=C0103
                         name=None,
-                        access=None):
+                        access=None):  # pylint: disable=C0103
     '''
     Return ec2_credentials for a user (keystone ec2-credentials-get)
 
@@ -90,21 +92,23 @@
     if name:
         for user in kstone.users.list():
             if user.name == name:
-                user_id = user.id
+                user_id = user.id  # pylint: disable=C0103
                 continue
     if not user_id:
         return {'Error': 'Unable to resolve user id'}
     if not access:
         return {'Error': 'Access key is required'}
     ec2_credentials = kstone.ec2.get(user_id=user_id, access=access)
-    ret[ec2_credentials.user_id] = {'user_id': ec2_credentials.user_id,
-                                    'tenant': ec2_credentials.tenant_id,
-                                    'access': ec2_credentials.access,
-                                    'secret': ec2_credentials.secret}
+    ret[ec2_credentials.user_id] = {
+            'user_id': ec2_credentials.user_id,
+            'tenant': ec2_credentials.tenant_id,
+            'access': ec2_credentials.access,
+            'secret': ec2_credentials.secret,
+            }
     return ret
 
 
-def ec2_credentials_list(user_id=None, name=None):
+def ec2_credentials_list(user_id=None, name=None):  # pylint: disable=C0103
     '''
     Return a list of ec2_credentials for a specific user (keystone ec2-credentials-list)
 
@@ -121,15 +125,17 @@
     if name:
         for user in kstone.users.list():
             if user.name == name:
-                user_id = user.id
+                user_id = user.id  # pylint: disable=C0103
                 continue
     if not user_id:
         return {'Error': 'Unable to resolve user id'}
     for ec2_credential in kstone.ec2.list(user_id):
-        ret[ec2_credential.user_id] = {'user_id': ec2_credential.user_id,
-                                       'tenant_id': ec2_credential.tenant_id,
-                                       'access': ec2_credential.access,
-                                       'secret': ec2_credential.secret}
+        ret[ec2_credential.user_id] = {
+                'user_id': ec2_credential.user_id,
+                'tenant_id': ec2_credential.tenant_id,
+                'access': ec2_credential.access,
+                'secret': ec2_credential.secret,
+                }
     return ret
 
 
@@ -160,16 +166,18 @@
     kstone = auth()
     ret = {}
     for endpoint in kstone.endpoints.list():
-        ret[endpoint.id] = {'id': endpoint.id,
-                            'region': endpoint.region,
-                            'adminurl': endpoint.adminurl,
-                            'internalurl': endpoint.internalurl,
-                            'publicurl': endpoint.publicurl,
-                            'service_id': endpoint.service_id}
+        ret[endpoint.id] = {
+                'id': endpoint.id,
+                'region': endpoint.region,
+                'adminurl': endpoint.adminurl,
+                'internalurl': endpoint.internalurl,
+                'publicurl': endpoint.publicurl,
+                'service_id': endpoint.service_id,
+                }
     return ret
 
 
-def role_get(role_id=None, name=None):
+def role_get(id=None, name=None):  # pylint: disable=C0103
     '''
     Return a specific roles (keystone role-get)
 
@@ -186,13 +194,15 @@
     if name:
         for role in kstone.roles.list():
             if role.name == name:
-                role_id = role.id
+                id = role.id  # pylint: disable=C0103
                 continue
-    if not role_id:
+    if not id:
         return {'Error': 'Unable to resolve role id'}
-    role = kstone.roles.get(role_id)
-    ret[role.name] = {'id': role.id,
-                      'name': role.name}
+    role = kstone.roles.get(id)
+    ret[role.name] = {
+            'id': role.id,
+            'name': role.name,
+            }
     return ret
 
 
@@ -209,12 +219,14 @@
     kstone = auth()
     ret = {}
     for role in kstone.roles.list():
-        ret[role.name] = {'id': role.id,
-                          'name': role.name}
+        ret[role.name] = {
+                'id': role.id,
+                'name': role.name,
+                }
     return ret
 
 
-def service_get(service_id=None, name=None):
+def service_get(id=None, name=None):  # pylint: disable=C0103
     '''
     Return a specific services (keystone service-get)
 
@@ -231,15 +243,17 @@
     if name:
         for service in kstone.services.list():
             if service.name == name:
-                service_id = service.id
+                id = service.id  # pylint: disable=C0103
                 continue
-    if not service_id:
+    if not id:
         return {'Error': 'Unable to resolve service id'}
-    service = kstone.services.get(service_id)
-    ret[service.name] = {'id': service.id,
-                         'name': service.name,
-                         'type': service.type,
-                         'description': service.description}
+    service = kstone.services.get(id)
+    ret[service.name] = {
+            'id': service.id,
+            'name': service.name,
+            'type': service.type,
+            'description': service.description,
+            }
     return ret
 
 
@@ -256,14 +270,16 @@
     kstone = auth()
     ret = {}
     for service in kstone.services.list():
-        ret[service.name] = {'id': service.id,
-                             'name': service.name,
-                             'description': service.description,
-                             'type': service.type}
+        ret[service.name] = {
+                'id': service.id,
+                'name': service.name,
+                'description': service.description,
+                'type': service.type,
+                }
     return ret
 
 
-def tenant_get(tenant_id=None, name=None):
+def tenant_get(id=None, name=None):  # pylint: disable=C0103
     '''
     Return a specific tenants (keystone tenant-get)
 
@@ -280,15 +296,17 @@
     if name:
         for tenant in kstone.tenants.list():
             if tenant.name == name:
-                tenant_id = tenant.id
+                id = tenant.id  # pylint: disable=C0103
                 continue
-    if not tenant_id:
+    if not id:
         return {'Error': 'Unable to resolve tenant id'}
-    tenant = kstone.tenants.get(tenant_id)
-    ret[tenant.name] = {'id': tenant.id,
-                        'name': tenant.name,
-                        'description': tenant.description,
-                        'enabled': tenant.enabled}
+    tenant = kstone.tenants.get(id)
+    ret[tenant.name] = {
+            'id': tenant.id,
+            'name': tenant.name,
+            'description': tenant.description,
+            'enabled': tenant.enabled,
+            }
     return ret
 
 
@@ -305,10 +323,12 @@
     kstone = auth()
     ret = {}
     for tenant in kstone.tenants.list():
-        ret[tenant.name] = {'id': tenant.id,
-                            'name': tenant.name,
-                            'description': tenant.description,
-                            'enabled': tenant.enabled}
+        ret[tenant.name] = {
+                'id': tenant.id,
+                'name': tenant.name,
+                'description': tenant.description,
+                'enabled': tenant.enabled,
+                }
     return ret
 
 
@@ -324,10 +344,12 @@
     '''
     kstone = auth()
     token = kstone.service_catalog.get_token()
-    return {'id': token['id'],
+    return {
+            'id': token['id'],
             'expires': token['expires'],
             'user_id': token['user_id'],
-            'tenant_id': token['tenant_id']}
+            'tenant_id': token['tenant_id'],
+            }
 
 
 def user_list():
@@ -343,15 +365,17 @@
     kstone = auth()
     ret = {}
     for user in kstone.users.list():
-        ret[user.name] = {'id': user.id,
-                          'name': user.name,
-                          'email': user.email,
-                          'enabled': user.enabled,
-                          'tenant_id': user.tenantId}
+        ret[user.name] = {
+                'id': user.id,
+                'name': user.name,
+                'email': user.email,
+                'enabled': user.enabled,
+                'tenant_id': user.tenantId,
+                }
     return ret
 
 
-def user_get(user_id=None, name=None):
+def user_get(user_id=None, name=None):  # pylint: disable=C0103
     '''
     Return a specific users (keystone user-get)
 
@@ -368,16 +392,18 @@
     if name:
         for user in kstone.users.list():
             if user.name == name:
-                user_id = user.id
+                user_id = user.id  # pylint: disable=C0103
                 continue
     if not user_id:
         return {'Error': 'Unable to resolve user id'}
     user = kstone.users.get(user_id)
-    ret[user.name] = {'id': user.id,
-                      'name': user.name,
-                      'email': user.email,
-                      'enabled': user.enabled,
-                      'tenant_id': user.tenantId}
+    ret[user.name] = {
+            'id': user.id,
+            'name': user.name,
+            'email': user.email,
+            'enabled': user.enabled,
+            'tenant_id': user.tenantId,
+            }
     return ret
 
 
@@ -392,15 +418,17 @@
         salt '*' keystone.user_create name=jack password=zero email=jack@halloweentown.org tenant_id=a28a7b5a999a455f84b1f5210264375e enabled=True
     '''
     kstone = auth()
-    item = kstone.users.create(name=name,
-                               password=password,
-                               email=email,
-                               tenant_id=tenant_id,
-                               enabled=enabled)
+    item = kstone.users.create(
+        name=name,
+        password=password,
+        email=email,
+        tenant_id=tenant_id,
+        enabled=enabled,
+        )
     return user_get(item.id)
 
 
-def user_delete(user_id=None, name=None):
+def user_delete(user_id=None, name=None):  # pylint: disable=C0103
     '''
     Delete a user (keystone user-delete)
 
@@ -416,7 +444,7 @@
     if name:
         for user in kstone.users.list():
             if user.name == name:
-                user_id = user.id
+                user_id = user.id  # pylint: disable=C0103
                 continue
     if not user_id:
         return {'Error': 'Unable to resolve user id'}
@@ -427,10 +455,10 @@
     return ret
 
 
-def user_update(user_id=None,
+def user_update(user_id=None,        # pylint: disable=C0103
                 name=None,
                 email=None,
-                enabled=None):
+                enabled=None):  # pylint: disable=C0103
     '''
     Update a user's information (keystone user-update)
     The following fields may be updated: name, email, enabled.
@@ -451,9 +479,9 @@
     return ret
 
 
-def user_password_update(user_id=None,
+def user_password_update(user_id=None,         # pylint: disable=C0103
                          name=None,
-                         password=None):
+                         password=None):  # pylint: disable=C0103
     '''
     Update a user's password (keystone user-password-update)
 
@@ -469,7 +497,7 @@
     if name:
         for user in kstone.users.list():
             if user.name == name:
-                user_id = user.id
+                user_id = user.id  # pylint: disable=C0103
                 continue
     if not user_id:
         return {'Error': 'Unable to resolve user id'}
@@ -511,10 +539,12 @@
     if not user_id and not tenant_id:
         return {'Error': 'Unable to resolve user or tenant id'}
     for role in kstone.roles.roles_for_user(user=user_id, tenant=tenant_id):
-        ret[role.name] = {'id': role.id,
-                          'name': role.name,
-                          'user_id': user_id,
-                          'tenant_id': tenant_id}
+        ret[role.name] = {
+                'id': role.id,
+                'name': role.name,
+                'user_id': user_id,
+                'tenant_id': tenant_id,
+                }
     return ret
 
 
@@ -561,4 +591,4 @@
     #discover            Discover Keystone servers and show authentication
     #                    protocols and
     #bootstrap           Grants a new role to a new user on a new tenant, after
-    #                    creating each.
+    #                    creating each.
\ No newline at end of file
