--- cases/repo_4fbf275_81f385c/merge_api.py	2026-03-28 21:11:19
+++ cases/repo_4fbf275_81f385c/llm_gemini_api.py	2026-06-09 12:09:35
@@ -84,7 +84,7 @@
 
     ```
     >>> import httpx
-    >>> response = httpx.request('GET', 'https://httpbin.org/get')
+    >>> response = httpx.request('GET', '[https://httpbin.org/get](https://httpbin.org/get)')
     >>> response
     <Response [200 OK]>
     ```
@@ -142,12 +142,7 @@
     [0]: /quickstart#streaming-responses
     """
     with Client(
-        cookies=cookies,
-        proxies=proxies,
-        cert=cert,
-        verify=verify,
-        timeout=timeout,
-        trust_env=trust_env,
+        cookies=cookies, proxies=proxies, cert=cert, verify=verify, timeout=timeout, trust_env=trust_env
     ) as client:
         with client.stream(
             method=method,
@@ -442,4 +437,4 @@
         verify=verify,
         timeout=timeout,
         trust_env=trust_env,
-    )
+    )
\ No newline at end of file
