diff --git a/ppdet/modeling/heads/ppyoloe_head.py b/ppdet/modeling/heads/ppyoloe_head.py
index 31e7590a0..01d7e9174 100644
--- a/ppdet/modeling/heads/ppyoloe_head.py
+++ b/ppdet/modeling/heads/ppyoloe_head.py
@@ -132,8 +132,8 @@ class PPYOLOEHead(nn.Layer):
 
         if self.eval_size:
             anchor_points, stride_tensor = self._generate_anchors()
-            self.register_buffer('anchor_points', anchor_points)
-            self.register_buffer('stride_tensor', stride_tensor)
+            self.anchor_points = anchor_points
+            self.stride_tensor = stride_tensor
 
     def forward_train(self, feats, targets):
         anchors, anchor_points, num_anchors_list, stride_tensor = \
