--- cases/repo_ff5e6e5_b07572e/merge_setup.py	2026-03-01 19:30:51.636985500 -0500
+++ cases/repo_ff5e6e5_b07572e/llm_setup.py	2026-03-03 12:08:54.454660200 -0500
@@ -51,8 +51,8 @@
     entry_points={
         'console_scripts': [
             'pre-commit = pre_commit.main:main',
-            'pre-commit-validate-config = pre_commit.clientlib.validate_config:run',  # noqa
-            'pre-commit-validate-manifest = pre_commit.clientlib.validate_manifest:run',  # noqa
+            'pre-commit-validate-config = pre_commit.clientlib.validate_config:run',
+            'pre-commit-validate-manifest = pre_commit.clientlib.validate_manifest:run',
         ],
     },
 )
