--- cases/repo_057203e_f305d2d/merge_data.py	2026-03-03 20:21:37
+++ cases/repo_057203e_f305d2d/llm_gemini_data.py	2026-06-05 12:45:13
@@ -1563,8 +1563,8 @@
                     summary = tf.Summary()
                     summary.value.extend(
                         [tf.Summary.Value(
-                            tag="bit_per_dim",
-                            simple_value=bit_per_dim)])
+                             tag="bit_per_dim",
+                             simple_value=bit_per_dim)])
                     summary_writer.add_summary(summary, current_step)
 
                     if return_val:
@@ -1633,4 +1633,4 @@
                  traindir=FLAGS.traindir, subset=FLAGS.mode)
 
 if __name__ == "__main__":
-    tf.app.run()
+    tf.app.run()
\ No newline at end of file
