--- cases/repo_f35fec0_ddc5cd8/merge_flib.py	2026-03-16 15:52:49
+++ cases/repo_f35fec0_ddc5cd8/llm_flib.py	2026-03-17 11:17:07
@@ -2203,7 +2203,7 @@
             headers (dict): headers
             noLog (bool): do not log request
             postData (str): HTTP POST data
-            disableContentEncoding (bool): do not UTF-8 encode response body
+            disableContentEncoding (bool): do not mangle
             sizeLimit (int): size threshold
             headOnly (bool): use HTTP HEAD method
             verify (bool): use HTTPS SSL/TLS verification
@@ -2571,4 +2571,4 @@
 
         return None
 
-# end of SpiderFoot class
+# end of SpiderFoot class
\ No newline at end of file
