--- cases/repo_829a393_f262e33/merge_utility.py	2026-04-05 13:06:33
+++ cases/repo_829a393_f262e33/llm_utility.py	2026-04-05 13:10:09
@@ -210,22 +210,22 @@
                 "nearest_interp_v2_0.tmp_0": [1, 256, 40, 40]
             }
             min_pact_shape = {
-                "nearest_interp_v2_26.tmp_0": [1, 256, 20, 20],
-                "nearest_interp_v2_27.tmp_0": [1, 64, 20, 20],
-                "nearest_interp_v2_28.tmp_0": [1, 64, 20, 20],
-                "nearest_interp_v2_29.tmp_0": [1, 64, 20, 20]
+                "nearest_interp_v2_26.tmp_0":[1,256,20,20],
+                "nearest_interp_v2_27.tmp_0":[1,64,20,20],
+                "nearest_interp_v2_28.tmp_0":[1,64,20,20],
+                "nearest_interp_v2_29.tmp_0":[1,64,20,20]
             }
             max_pact_shape = {
-                "nearest_interp_v2_26.tmp_0": [1, 256, 400, 400],
-                "nearest_interp_v2_27.tmp_0": [1, 64, 400, 400],
-                "nearest_interp_v2_28.tmp_0": [1, 64, 400, 400],
-                "nearest_interp_v2_29.tmp_0": [1, 64, 400, 400]
+                "nearest_interp_v2_26.tmp_0":[1,256,400,400],
+                "nearest_interp_v2_27.tmp_0":[1,64,400,400],
+                "nearest_interp_v2_28.tmp_0":[1,64,400,400],
+                "nearest_interp_v2_29.tmp_0":[1,64,400,400]
             }
             opt_pact_shape = {
-                "nearest_interp_v2_26.tmp_0": [1, 256, 160, 160],
-                "nearest_interp_v2_27.tmp_0": [1, 64, 160, 160],
-                "nearest_interp_v2_28.tmp_0": [1, 64, 160, 160],
-                "nearest_interp_v2_29.tmp_0": [1, 64, 160, 160]
+                "nearest_interp_v2_26.tmp_0":[1,256,160,160],
+                "nearest_interp_v2_27.tmp_0":[1,64,160,160],
+                "nearest_interp_v2_28.tmp_0":[1,64,160,160],
+                "nearest_interp_v2_29.tmp_0":[1,64,160,160]
             }
             min_input_shape.update(min_pact_shape)
             max_input_shape.update(max_pact_shape)
@@ -564,4 +564,4 @@
 
 
 if __name__ == '__main__':
-    pass
+    pass
\ No newline at end of file
