diff --git a/salt/modules/keystone.py b/salt/modules/keystone.py
index 93d594ab63..51e9fdfa6c 100644
--- a/salt/modules/keystone.py
+++ b/salt/modules/keystone.py
@@ -45,37 +45,35 @@ def auth():
 
     Only intended to be used within Keystone-enabled modules
     '''
-    user = __salt__['config.option']('keystone.user')
-    password = __salt__['config.option']('keystone.password')
-    tenant = __salt__['config.option']('keystone.tenant')
+    user = __salt__['config.option']('keystone.user', 'admin')
+    password = __salt__['config.option']('keystone.password', 'ADMIN')
+    tenant = __salt__['config.option']('keystone.tenant', 'admin')
     tenant_id = __salt__['config.option']('keystone.tenant_id')
-    auth_url = __salt__['config.option']('keystone.auth_url')
-    insecure = __salt__['config.option']('keystone.insecure')
-    token = __salt__['config.option']('keystone.token')
-    endpoint = __salt__['config.option']('keystone.endpoint')
+    auth_url = __salt__['config.option']('keystone.auth_url',
+                                         'http://127.0.0.1:35357/v2.0/')
+    insecure = __salt__['config.option']('keystone.insecure', False)
+    token = __salt__['config.option']('keystone.token', 'ADMIN')
+    endpoint = __salt__['config.option']('keystone.endpoint',
+                                         'http://127.0.0.1:35357/v2.0')
 
     kwargs = {}
     if token:
-        kwargs = {
-                'token': token,
-                'endpoint': endpoint,
-                }
+        kwargs = {'token': token,
+                  'endpoint': endpoint}
     else:
-        kwargs = {
-                'username': user,
-                'password': password,
-                'tenant_name': tenant,
-                'tenant_id': tenant_id,
-                'auth_url': auth_url,
-                'insecure': insecure,
-                }
+        kwargs = {'username': user,
+                  'password': password,
+                  'tenant_name': tenant,
+                  'tenant_id': tenant_id,
+                  'auth_url': auth_url,
+                  'insecure': insecure}
 
     return client.Client(**kwargs)
 
 
-def ec2_credentials_get(id=None,       # pylint: disable=C0103
+def ec2_credentials_get(user_id=None,
                         name=None,
-                        access=None):  # pylint: disable=C0103
+                        access=None):
     '''
     Return ec2_credentials for a user (keystone ec2-credentials-get)
 
@@ -92,23 +90,21 @@ def ec2_credentials_get(id=None,       # pylint: disable=C0103
     if name:
         for user in kstone.users.list():
             if user.name == name:
-                id = user.id  # pylint: disable=C0103
+                user_id = user.id
                 continue
-    if not id:
+    if not user_id:
         return {'Error': 'Unable to resolve user id'}
     if not access:
         return {'Error': 'Access key is required'}
-    ec2_credentials = kstone.ec2.get(user_id=id, access=access)
-    ret[ec2_credentials.user_id] = {
-            'user_id': ec2_credentials.user_id,
-            'tenant': ec2_credentials.tenant_id,
-            'access': ec2_credentials.access,
-            'secret': ec2_credentials.secret,
-            }
+    ec2_credentials = kstone.ec2.get(user_id=user_id, access=access)
+    ret[ec2_credentials.user_id] = {'user_id': ec2_credentials.user_id,
+                                    'tenant': ec2_credentials.tenant_id,
+                                    'access': ec2_credentials.access,
+                                    'secret': ec2_credentials.secret}
     return ret
 
 
-def ec2_credentials_list(id=None, name=None):  # pylint: disable=C0103
+def ec2_credentials_list(user_id=None, name=None):
     '''
     Return a list of ec2_credentials for a specific user (keystone ec2-credentials-list)
 
@@ -125,17 +121,15 @@ def ec2_credentials_list(id=None, name=None):  # pylint: disable=C0103
     if name:
         for user in kstone.users.list():
             if user.name == name:
-                id = user.id  # pylint: disable=C0103
+                user_id = user.id
                 continue
-    if not id:
+    if not user_id:
         return {'Error': 'Unable to resolve user id'}
-    for ec2_credential in kstone.ec2.list(id):
-        ret[ec2_credential.user_id] = {
-                'user_id': ec2_credential.user_id,
-                'tenant_id': ec2_credential.tenant_id,
-                'access': ec2_credential.access,
-                'secret': ec2_credential.secret,
-                }
+    for ec2_credential in kstone.ec2.list(user_id):
+        ret[ec2_credential.user_id] = {'user_id': ec2_credential.user_id,
+                                       'tenant_id': ec2_credential.tenant_id,
+                                       'access': ec2_credential.access,
+                                       'secret': ec2_credential.secret}
     return ret
 
 
@@ -166,18 +160,16 @@ def endpoint_list():
     kstone = auth()
     ret = {}
     for endpoint in kstone.endpoints.list():
-        ret[endpoint.id] = {
-                'id': endpoint.id,
-                'region': endpoint.region,
-                'adminurl': endpoint.adminurl,
-                'internalurl': endpoint.internalurl,
-                'publicurl': endpoint.publicurl,
-                'service_id': endpoint.service_id,
-                }
+        ret[endpoint.id] = {'id': endpoint.id,
+                            'region': endpoint.region,
+                            'adminurl': endpoint.adminurl,
+                            'internalurl': endpoint.internalurl,
+                            'publicurl': endpoint.publicurl,
+                            'service_id': endpoint.service_id}
     return ret
 
 
-def role_get(id=None, name=None):  # pylint: disable=C0103
+def role_get(role_id=None, name=None):
     '''
     Return a specific roles (keystone role-get)
 
@@ -194,15 +186,13 @@ def role_get(id=None, name=None):  # pylint: disable=C0103
     if name:
         for role in kstone.roles.list():
             if role.name == name:
-                id = role.id  # pylint: disable=C0103
+                role_id = role.id
                 continue
-    if not id:
+    if not role_id:
         return {'Error': 'Unable to resolve role id'}
-    role = kstone.roles.get(id)
-    ret[role.name] = {
-            'id': role.id,
-            'name': role.name,
-            }
+    role = kstone.roles.get(role_id)
+    ret[role.name] = {'id': role.id,
+                      'name': role.name}
     return ret
 
 
@@ -219,14 +209,12 @@ def role_list():
     kstone = auth()
     ret = {}
     for role in kstone.roles.list():
-        ret[role.name] = {
-                'id': role.id,
-                'name': role.name,
-                }
+        ret[role.name] = {'id': role.id,
+                          'name': role.name}
     return ret
 
 
-def service_get(id=None, name=None):  # pylint: disable=C0103
+def service_get(service_id=None, name=None):
     '''
     Return a specific services (keystone service-get)
 
@@ -243,17 +231,15 @@ def service_get(id=None, name=None):  # pylint: disable=C0103
     if name:
         for service in kstone.services.list():
             if service.name == name:
-                id = service.id  # pylint: disable=C0103
+                service_id = service.id
                 continue
-    if not id:
+    if not service_id:
         return {'Error': 'Unable to resolve service id'}
-    service = kstone.services.get(id)
-    ret[service.name] = {
-            'id': service.id,
-            'name': service.name,
-            'type': service.type,
-            'description': service.description,
-            }
+    service = kstone.services.get(service_id)
+    ret[service.name] = {'id': service.id,
+                         'name': service.name,
+                         'type': service.type,
+                         'description': service.description}
     return ret
 
 
@@ -270,16 +256,14 @@ def service_list():
     kstone = auth()
     ret = {}
     for service in kstone.services.list():
-        ret[service.name] = {
-                'id': service.id,
-                'name': service.name,
-                'description': service.description,
-                'type': service.type,
-                }
+        ret[service.name] = {'id': service.id,
+                             'name': service.name,
+                             'description': service.description,
+                             'type': service.type}
     return ret
 
 
-def tenant_get(id=None, name=None):  # pylint: disable=C0103
+def tenant_get(tenant_id=None, name=None):
     '''
     Return a specific tenants (keystone tenant-get)
 
@@ -296,17 +280,15 @@ def tenant_get(id=None, name=None):  # pylint: disable=C0103
     if name:
         for tenant in kstone.tenants.list():
             if tenant.name == name:
-                id = tenant.id  # pylint: disable=C0103
+                tenant_id = tenant.id
                 continue
-    if not id:
+    if not tenant_id:
         return {'Error': 'Unable to resolve tenant id'}
-    tenant = kstone.tenants.get(id)
-    ret[tenant.name] = {
-            'id': tenant.id,
-            'name': tenant.name,
-            'description': tenant.description,
-            'enabled': tenant.enabled,
-            }
+    tenant = kstone.tenants.get(tenant_id)
+    ret[tenant.name] = {'id': tenant.id,
+                        'name': tenant.name,
+                        'description': tenant.description,
+                        'enabled': tenant.enabled}
     return ret
 
 
@@ -323,12 +305,10 @@ def tenant_list():
     kstone = auth()
     ret = {}
     for tenant in kstone.tenants.list():
-        ret[tenant.name] = {
-                'id': tenant.id,
-                'name': tenant.name,
-                'description': tenant.description,
-                'enabled': tenant.enabled,
-                }
+        ret[tenant.name] = {'id': tenant.id,
+                            'name': tenant.name,
+                            'description': tenant.description,
+                            'enabled': tenant.enabled}
     return ret
 
 
@@ -344,12 +324,10 @@ def token_get():
     '''
     kstone = auth()
     token = kstone.service_catalog.get_token()
-    return {
-            'id': token['id'],
+    return {'id': token['id'],
             'expires': token['expires'],
             'user_id': token['user_id'],
-            'tenant_id': token['tenant_id'],
-            }
+            'tenant_id': token['tenant_id']}
 
 
 def user_list():
@@ -365,17 +343,15 @@ def user_list():
     kstone = auth()
     ret = {}
     for user in kstone.users.list():
-        ret[user.name] = {
-                'id': user.id,
-                'name': user.name,
-                'email': user.email,
-                'enabled': user.enabled,
-                'tenant_id': user.tenantId,
-                }
+        ret[user.name] = {'id': user.id,
+                          'name': user.name,
+                          'email': user.email,
+                          'enabled': user.enabled,
+                          'tenant_id': user.tenantId}
     return ret
 
 
-def user_get(id=None, name=None):  # pylint: disable=C0103
+def user_get(user_id=None, name=None):
     '''
     Return a specific users (keystone user-get)
 
@@ -392,18 +368,16 @@ def user_get(id=None, name=None):  # pylint: disable=C0103
     if name:
         for user in kstone.users.list():
             if user.name == name:
-                id = user.id  # pylint: disable=C0103
+                user_id = user.id
                 continue
-    if not id:
+    if not user_id:
         return {'Error': 'Unable to resolve user id'}
-    user = kstone.users.get(id)
-    ret[user.name] = {
-            'id': user.id,
-            'name': user.name,
-            'email': user.email,
-            'enabled': user.enabled,
-            'tenant_id': user.tenantId,
-            }
+    user = kstone.users.get(user_id)
+    ret[user.name] = {'id': user.id,
+                      'name': user.name,
+                      'email': user.email,
+                      'enabled': user.enabled,
+                      'tenant_id': user.tenantId}
     return ret
 
 
@@ -418,17 +392,15 @@ def user_create(name, password, email, tenant_id=None, enabled=True):
         salt '*' keystone.user_create name=jack password=zero email=jack@halloweentown.org tenant_id=a28a7b5a999a455f84b1f5210264375e enabled=True
     '''
     kstone = auth()
-    item = kstone.users.create(
-        name=name,
-        password=password,
-        email=email,
-        tenant_id=tenant_id,
-        enabled=enabled,
-        )
+    item = kstone.users.create(name=name,
+                               password=password,
+                               email=email,
+                               tenant_id=tenant_id,
+                               enabled=enabled)
     return user_get(item.id)
 
 
-def user_delete(id=None, name=None):  # pylint: disable=C0103
+def user_delete(user_id=None, name=None):
     '''
     Delete a user (keystone user-delete)
 
@@ -444,21 +416,21 @@ def user_delete(id=None, name=None):  # pylint: disable=C0103
     if name:
         for user in kstone.users.list():
             if user.name == name:
-                id = user.id  # pylint: disable=C0103
+                user_id = user.id
                 continue
-    if not id:
+    if not user_id:
         return {'Error': 'Unable to resolve user id'}
-    kstone.users.delete(id)
-    ret = 'User ID {0} deleted'.format(id)
+    kstone.users.delete(user_id)
+    ret = 'User ID {0} deleted'.format(user_id)
     if name:
         ret += ' ({0})'.format(name)
     return ret
 
 
-def user_update(id=None,        # pylint: disable=C0103
+def user_update(user_id=None,
                 name=None,
                 email=None,
-                enabled=None):  # pylint: disable=C0103
+                enabled=None):
     '''
     Update a user's information (keystone user-update)
     The following fields may be updated: name, email, enabled.
@@ -472,16 +444,16 @@ def user_update(id=None,        # pylint: disable=C0103
         salt '*' keystone.user_update c965f79c4f864eaaa9c3b41904e67082 name=newname email=newemail@domain.com
     '''
     kstone = auth()
-    if not id:
+    if not user_id:
         return {'Error': 'Unable to resolve user id'}
-    kstone.users.update(user=id, name=name, email=email, enabled=enabled)
-    ret = 'Info updated for user ID {0}'.format(id)
+    kstone.users.update(user=user_id, name=name, email=email, enabled=enabled)
+    ret = 'Info updated for user ID {0}'.format(user_id)
     return ret
 
 
-def user_password_update(id=None,         # pylint: disable=C0103
+def user_password_update(user_id=None,
                          name=None,
-                         password=None):  # pylint: disable=C0103
+                         password=None):
     '''
     Update a user's password (keystone user-password-update)
 
@@ -497,12 +469,12 @@ def user_password_update(id=None,         # pylint: disable=C0103
     if name:
         for user in kstone.users.list():
             if user.name == name:
-                id = user.id  # pylint: disable=C0103
+                user_id = user.id
                 continue
-    if not id:
+    if not user_id:
         return {'Error': 'Unable to resolve user id'}
-    kstone.users.update_password(user=id, password=password)
-    ret = 'Password updated for user ID {0}'.format(id)
+    kstone.users.update_password(user=user_id, password=password)
+    ret = 'Password updated for user ID {0}'.format(user_id)
     if name:
         ret += ' ({0})'.format(name)
     return ret
@@ -539,12 +511,10 @@ def user_role_list(user_id=None,
     if not user_id and not tenant_id:
         return {'Error': 'Unable to resolve user or tenant id'}
     for role in kstone.roles.roles_for_user(user=user_id, tenant=tenant_id):
-        ret[role.name] = {
-                'id': role.id,
-                'name': role.name,
-                'user_id': user_id,
-                'tenant_id': tenant_id,
-                }
+        ret[role.name] = {'id': role.id,
+                          'name': role.name,
+                          'user_id': user_id,
+                          'tenant_id': tenant_id}
     return ret
 
 
